src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -140,8 +146,29 @@ internal IEnumerable<NodeSearchElement> GetMatchingSearchElements()
             //check if the input port return type is in the skipped input types list
             if (skippedInputTypes.Any(s => s == inputPortType))
             {
+                switch (inputPortType)
+                {
+                    case "int":
+                        DefaultResults = AllDefaultResults.Where(e => e.Name == "Number Slider" || e.Name == "Integer Slider").ToList();
+                        break;
+                    case "double":
+                        DefaultResults = AllDefaultResults.Where(e => e.Name == "Number Slider" || e.Name == "Integer Slider").ToList();
+                        break;
+                    case "string":
+                        DefaultResults = AllDefaultResults.Where(e => e.Name == "String").ToList();
+                        break;
+                    case "bool":
+                        DefaultResults = AllDefaultResults.Where(e => e.Name == "Boolean").ToList();
+                        break;
+                    default:
+                        DefaultResults = AllDefaultResults;
+                        break;
+                }
                 return elements;
             }
+            else {
+                DefaultResults = AllDefaultResults;
```

Comment:
@zeusongit Why are we modifying `DefaultResults` (class property) in this Utility or API function? I think previously we are only returning elements. I would like to be consistent with that. Without that, I feel we do not need to create a new property `AllDefaultResults`, what do you think?

Reply:
That was my initial intention, but for doing that we will have to query the default nodes from the list of all the nodes, which could be a time consuming process for a trivial task. So searching a type out of a list of five nodes made more sense to me.
Or 
an alternative approach that I thought of was to define a dictionary of those 5 default types one time and assign those from the dictionary into the elements collection based on the input port.

Reply:
yeah, this feels pretty weird 😉 - I think a dictionary with the proper list is more straightforward - why keep state if you don't have to.

Reply:
@zeusongit What about touch `InitializeDefaultAutoCompleteCandidates()` for those input types, make the default results correct in the first place so we do not need to touch this API?

Reply:
@QilongTang InitializeDefaultAutoCompleteCandidates() is only called once when the constructor is called, not every time the user clicks on an input port.

Reply:
@zeusongit I see, then produce a dictionary sounds proper. Would you update the code following what you described earlier?